src/Umbraco.Core/Media/MediaHelper.cs

```diff
@@ -0,0 +1,280 @@
+            if (string.IsNullOrWhiteSpace(filename)) throw new ArgumentException("Null or empty.", "filename");
+            if (filestream == null) throw new ArgumentNullException("filestream");
+
+            // clear the old file, if any
+            var fs = FileSystem;
+            if (string.IsNullOrWhiteSpace(oldpath))
+                fs.DeleteFile(oldpath, true);
+
+            // get the filepath, store the data
+            // use oldpath as "prevpath" to try and reuse the folder, in original number-based scheme
+            var filepath = GetMediaPath(filename, oldpath, content.Key, propertyType.Key);
+            fs.AddFile(filepath, filestream);
+            return filepath;
+        }
+
+        /// <summary>
+        /// Clears a media file.
+        /// </summary>
+        /// <param name="filepath">The filesystem-relative path to the media file.</param>
+        public static void DeleteFile(string filepath)
+        {
+            FileSystem.DeleteFile(filepath, true);
+        }
+
+        /// <summary>
+        /// Copies a media file.
+        /// </summary>
+        /// <param name="content">The content item owning the copy of the media file.</param>
+        /// <param name="propertyType">The property type owning the copy of the media file.</param>
+        /// <param name="sourcepath">The filesystem-relative path to the source media file.</param>
+        /// <returns>The filesystem-relative path to the copy of the media file.</returns>
+        public static string CopyFile(IContentBase content, PropertyType propertyType, string sourcepath)
+        {
+            if (content == null) throw new ArgumentNullException("content");
+            if (propertyType == null) throw new ArgumentNullException("propertyType");
+            if (string.IsNullOrWhiteSpace(sourcepath)) throw new ArgumentException("Null or empty.", "sourcepath");
+
+            // ensure we have a file to copy
+            var fs = FileSystem;
+            if (fs.FileExists(sourcepath) == false) return null;
+
+            // get the filepath
+            var filename = Path.GetFileName(sourcepath);
+            var filepath = GetMediaPath(filename, content.Key, propertyType.Key);
+            fs.CopyFile(sourcepath, filepath);
+            fs.CopyThumbnails(sourcepath, filepath);
+            return filepath;
+        }
+
+        /// <summary>
+        /// Gets or creates a property for a content item.
+        /// </summary>
+        /// <param name="content">The content item.</param>
+        /// <param name="propertyTypeAlias">The property type alias.</param>
+        /// <returns>The property.</returns>
+        private static Property GetProperty(IContentBase content, string propertyTypeAlias)
+        {
+            var property = content.Properties.FirstOrDefault(x => x.Alias.InvariantEquals(propertyTypeAlias));
+            if (property != null) return property;
+
+            var propertyType = content.GetContentType().CompositionPropertyTypes
```

Comment:
This usage: `content.GetContentType()` is an extension method isn't it? We should never ourselves use any extension method that requires the ApplicationContext.Current singleton, this make things un-testable. If the ApplicationContext or IContentService is required for these methods then they need to be passed in to the ctor of this class.